shell/browser/serial/serial_chooser_controller.cc

```diff
@@ -140,6 +147,28 @@ api::Session* SerialChooserController::GetSession() {
   return api::Session::FromBrowserContext(web_contents_->GetBrowserContext());
 }
 
+void SerialChooserController::GetDevices() {
+  if (IsWirelessSerialPortOnly()) {
+    if (!adapter_) {
+      BluetoothAdapterFactory::Get()->GetAdapter(base::BindOnce(
+          &SerialChooserController::OnGetAdapter, weak_factory_.GetWeakPtr(),
+          base::BindOnce(&SerialChooserController::GetDevices,
+                         weak_factory_.GetWeakPtr())));
+      return;
+    }
+  }
+
+  chooser_context_->GetPortManager()->GetDevices(base::BindOnce(
+      &SerialChooserController::OnGetDevices, weak_factory_.GetWeakPtr()));
+}
+
+void SerialChooserController::AdapterPoweredChanged(BluetoothAdapter* adapter,
+                                                    bool powered) {
+  if (powered) {
+    GetDevices();
+  }
```

Comment:
Asking for context: do we need to do anything when `powered` changes to `false` here? For example do we need to remove devices when powered is false?

Reply:
At least not per https://chromium-review.googlesource.com/c/chromium/src/+/5737296